fix:  Uncaught NotSupportedError in Reactjs  #8

Dunsin-cyber opened the pull request:
**Hi @bumi,**

This fixes #8.

The issue was caused by mutating `classList` inside the component’s constructor, which triggered a runtime error because the element hadn’t been attached to the DOM yet. I’ve moved that logic to `connectedCallback()`, which safely runs after the element is connected  that resolved the problem.

I spent most of yesterday working on the React wrapper. While it didn’t fix this particular issue, it’s still important for smooth integration with React projects. I’ve made a separate commit for that, which includes:

* The React wrapper (`simple-boost-react.ts`)
* A demo React project to test integration

I also took the opportunity to restructure a bit:

* `src/simple-boost.ts` is now split into `simple-boost.ts` and `simple-boost-react.ts`
* `src/index.ts` exports both
* I updated `package.json` and `rollup.config.js` to reflect this new structure and entry point

For anyone wanting to use the React wrapper, I’ve added documentation in
📄 `demos/react/README.md`
and a working example in
⚙️ `demos/react/app.tsx`

Let me know if anything should be tweaked!

bumi commented:
ah, great find! 
and yeah, the react component is great. Are there any breaking changes regarding the import? something people have to be aware of who already import it?

pavanjoshi914 commented on demos/react/README.md:
```diff
@@ -0,0 +1,52 @@
+# 🪄 SimpleBoost React Wrapper
```
can be improved a bit with step wise instructions

Dunsin-cyber replied:
alright

pavanjoshi914 commented:
I guess simpleboostbutton name is okay for me

On Wed, 6 Aug, 2025, 16:22 Dunsin, ***@***.***> wrote:

> *Dunsin-cyber* left a comment (getAlby/simple-boost#9)
> <https://github.com/getAlby/simple-boost/pull/9#issuecomment-3159549811>
>
> Alright @pavanjoshi914 <https://github.com/pavanjoshi914> , i'd make a
> push soon with the changes,
> a folow up question, should i change the ReactWrapper's name from
> SimpleBoostButton to simply BoostButton or it's okay that way
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/getAlby/simple-boost/pull/9#issuecomment-3159549811>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANKC3AQGOGU3NSHG6H4QM3T3MHM7DAVCNFSM6AAAAACCJENXWSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTCNJZGU2DSOBRGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Dunsin-cyber commented on src/simple-boost-react.ts:
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
alright

Dunsin-cyber commented on demos/react/.gitignore:
```diff
@@ -0,0 +1,24 @@
+!.vscode/extensions.json
+.idea
+.DS_Store
+*.suo
```
alright, got it.

pavanjoshi914 commented on demos/react/README.md:
````diff
@@ -0,0 +1,141 @@
+Then import it like this:
+
+```tsx
+import { SimpleBoostButton } from "simple-boost";
````
did you checked how that if such imports really work. also make sure that imports for existing users work

Dunsin-cyber replied:
so there are now two bundled outputs
- `simple-boost.bundled.js` and 
- `simple-boost-react.bundled.js`

<img width="893" height="461" alt="buid-summary-updated" src="https://github.com/user-attachments/assets/8a290de3-8788-4e30-ad02-f8cd0a8b5740" />

bumi replied:
what's all in the bundle that it gets 700kb+?

Dunsin-cyber replied:
i'm still trying to figure out why it added a few more kilobytes. 
i noticed when I bundled with npm, sometimes it gave 600 kb+, and when with yarn, it gave 700 kb+, but i'm on it.

bumi replied:
interesting. keep me posted.

Dunsin-cyber commented:
> can you also update the docs. regarding the import and component to us when in react. they are present in same repo. also some info in readme will be great!
> 
> actually readme can be improved a bit. when you see this readme on npm its still confusing a bit and missing key information and proper steps
> 
> https://getalby.github.io/simple-boost/


**@pavanjoshi914**
For the docs update, I’d like to propose simplifying how we handle code examples.

Right now, to show the code behind each button, we’re manually adding a bunch of classes to every `<code>` tag and structuring the HTML by hand. This works, but it’s verbose, harder to read, and tedious to maintain.

Since we’re already using Prism (the code display CDN), we can take advantage of its built-in parsing and styling instead. This means we could go from something like:

```html
class="language-html"><code class="language-html"><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>simple-boost</span> <span class="token attr-name">amount</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>2100<span class="token punctuation">"</span></span> <span class="token attr-name">address</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>hello@getalby.com<span class="token punctuation">"</span></span><span class="token punctuation">></span></span><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>simple-boost</span><span class="token punctuation">></span></span></code></pre>
```

…to something much cleaner:

```html
<pre class="language-html"><code class="language-html">
  &lt;simple-boost amount="2100" address="hello@getalby.com">&lt;/simple-boost>
</code></pre>
```

From there, we just import Prism’s JavaScript file downloaded from the site with all the styling in it, about 20KB in size and directly import it in  our HTML. It automatically applies the same syntax colors, formatting, and styles. No manual token spans or extra markup needed.

**Benefits:**

* **Same visual result** as now.
* **Much more readable** for contributors.
* **Significantly easier to maintain**  less manual HTML noise.
* **Future-proof**  any style changes from Prism will automatically apply.

I think this change would make the docs more approachable and reduce future editing overhead.

bumi commented:
great that react is optional and not automatically installed.

pavanjoshi914 commented on demos/react/README.md:
````diff
@@ -0,0 +1,141 @@
+1. Move back into the React demo folder:
+
+```bash
+cd demo/react
````
should be demos/react fix all instances

pavanjoshi914 commented on docs/prism.js:
also boost buttons in docs, doesn't render and run

pavanjoshi914 replied:
its not best practice to put everthing in same pr. maybe we should have done this migration separately

Dunsin-cyber replied:
okay, so if i'm getting you right, i should introduce these docs update in a different PR?
and keep the main PR simply for addng the react wrapper code

Dunsin-cyber commented:
> functionality works fine, docs broken, new instructions to use react wrapper still missing, updated package-lock.json missing
> 
> i am unable to commit to this pr. dunsin can you make those changes on your side and push it today?

alright, sure, i'd fix them and push today.
however, can you kindly help with the screenshot of the warnings that comes up when you run  ` npm run serve:docs`, 
the boost buttons also work on my end, maybe i'd test in a different environment.
also about this `new instructions to use react wrapper still missing,`, on which page is it missing, i added it to all the necessary pages like so
<img width="1366" height="768" alt="page-1" src="https://github.com/user-attachments/assets/5981c8f2-e5bb-482a-b26b-1c72ee6ecabb" />
<img width="1366" height="768" alt="page-2" src="https://github.com/user-attachments/assets/a5addce4-392d-4772-9681-91587a384742" />
<img width="1366" height="768" alt="page-3" src="https://github.com/user-attachments/assets/5ff761f8-193d-4dd9-9cd4-e8b26f26d461" />
and every other examples like fiat amount, no-confetti etc

pavanjoshi914 commented:
i followed root level readme to run docs. maybe you should update readme and scripts in package.json and also remove uncessary scripts related to web-dev-server. 

i did just npm run docs:serve then it worked. but after build and serve it failed (as suggested in readme)

pavanjoshi914 commented:
do we really need wds (web-dev-server) npm package? i guess it was used to build and serve docs as well as web component 

but when web component and docs worked only when i served them via vite. can you check and do cleanup? update scripts remove unrelated configs and dependencies?

pavanjoshi914 commented on package.json:
```diff
@@ -7,32 +7,41 @@
+      "import": "./dist/simple-boost.js"
+    },
+    "./react": {
+      "types": "./dist/siple-boost-react.d.ts",
```
you should always self review the changes before pushing
